src/visualizers/panels/BIPCodeEditor/BIPCodeEditorControl.js

```diff
@@ -0,0 +1,387 @@
+
+                    segmentId = addSegment('singleTransition', model.transitions[i].path, model.transitions[i]);
+
+                    parseResult = javaParser.checkForSingleFunction(
+                        segmentedDocument.segments[segmentId].value,
+                        null,
+                        'public',
+                        self._getSegmentOffset(segmentedDocument, segmentId));
+                    if (parseResult) {
+                        segmentedDocument.errors.push(parseResult);
+                    }
+                }
+
+                for (i = 0; i < model.guards.length; i += 1) {
+                    addSegment('singleGuardAnnotation', model.guards[i].path, model.guards[i], true);
+                    segmentId = addSegment('singleGuard', model.guards[i].path, model.guards[i]);
+                    parseResult = javaParser.checkForSingleFunction(
+                        segmentedDocument.segments[segmentId].value,
+                        'boolean',
+                        'public',
+                        self._getSegmentOffset(segmentedDocument, segmentId));
+                    if (parseResult) {
+                        segmentedDocument.errors.push(parseResult);
+                    }
+                }
+
+                addSegment('classEnd', model.path, model, true);
+
+                deferred.resolve(segmentedDocument);
+            })
+            .catch(deferred.reject);
+
+        return deferred.promise;
+    };
+
+    BIPCodeEditorControl.prototype._SaveDocument = function (changedSegments) {
+        var segment,
+            segmentId;
+
+        this._client.startTransaction();
+        for (segmentId in changedSegments) {
+            segment = changedSegments[segmentId];
+            segmentId = segmentId.split('*');
+
+            switch (segmentId[0]) {
+                case 'userImports':
+                    this._client.setAttribute(segmentId[1], 'forwards', segment);
+                    break;
+                case 'userDefinitions':
+                    this._client.setAttribute(segmentId[1], 'definitions', segment);
+                    break;
+                case 'userConstructors':
+                    this._client.setAttribute(segmentId[1], 'constructors', segment);
+                    break;
+                case 'singleTransition':
+                    this._client.setAttribute(segmentId[1], 'transitionMethod', segment);
+                    break;
+                case 'singleGuard':
+                    this._client.setAttribute(segmentId[1], 'guardMethod', segment);
+                    break;
+            }
+        }
+        this._client.completeTransaction();
+    };
+
+    /* * * * * * * * Node Event Handling * * * * * * * */
+    //TODO needs a better change management
+    BIPCodeEditorControl.prototype._eventCallback = function (events) {
+        var self = this;
+
+        if (events.length > 1) {
+            if (self._gatheringSegments) {
```

Comment:
Looks like this is always false.. Wouldn't that mean that the buildSegmentInfo is called multiple times since the territory contains all children?